code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -873,16 +873,17 @@
 	var/hrefpart = "<a href='?src=[REF(src)];track=[html_encode(namepart)]'>"
 	var/jobpart = "Unknown"
 
-	if (isliving(speaker))
-		var/mob/living/living_speaker = speaker
-		if(living_speaker.job)
-			jobpart = "[living_speaker.job]"
-	if (istype(speaker, /obj/effect/overlay/holo_pad_hologram))
-		var/obj/effect/overlay/holo_pad_hologram/holo = speaker
-		if(holo.Impersonation?.job)
-			jobpart = "[holo.Impersonation.job]"
-		else if(usr?.job) // not great, but AI holograms have no other usable ref
-			jobpart = "[usr.job]"
+	if(!HAS_TRAIT(speaker, TRAIT_UNKNOWN)) //don't fetch the speaker's job in case they have something that conseals their identity completely
```

Comment:
Maybe you can use ``get_id_name()`` instead?

Reply:
i'm not seeing how that would be better? it checks for the trait as well and we don't need the name at this point. couldn't it also lead to a false positive from a speaker just hiding their id and face but not their voice?

Reply:
What we want us the AI to hear people's names and jobs as they present themselves, not as they actually are, right?
I was thinking we can use ``get_id_name()`` to get their ID, then we check for job there. It checks for the trait there, and will return "Unknown" if they have it.

Reply:
that could shift the job fetch to the ID though, seems like a change in functionality. right now it pulls from the speaker directly, IC could be voice recognition? i might be confused